sessionctx/variable/session.go

```diff
@@ -887,7 +890,7 @@ func (s *SessionVars) RaiseWarningWhenMPPEnforced(warning string) {
 
 // CheckAndGetTxnScope will return the transaction scope we should use in the current session.
 func (s *SessionVars) CheckAndGetTxnScope() string {
-	if s.InRestrictedSQL {
+	if s.InRestrictedSQL || !s.EnableLocalTxn {
```

Comment:
Does it affect stale read? I see `CheckAndGetTxnScope()` used to initialize `TxnCtx`, and txnCtx.TxnScope is used in `handleAsOfAndReadTS`.

Reply:
Stale read bypass `CheckAndGetTxnScope`.